File: packages/db/src/schema/apiKey.ts
import {
  index,
  integer,
  pgTable,
  serial,
  timestamp,
  varchar,
} from "drizzle-orm/pg-core";

export const apiKey = pgTable(
  "api_key",
  {
    id: serial().primaryKey().notNull(),
    name: varchar({ length: 255 }).default("").notNull(),
    openUserId: varchar("open_user_id", { length: 32 }).default("").notNull(),
    key: varchar({ length: 255 }).notNull(),
    status: integer().default(0).notNull(),
    createdAt: timestamp("created_at", {
      withTimezone: true,
      mode: "date",
    }).defaultNow(),
    expiresAt: timestamp("expires_at", { withTimezone: true, mode: "date" }),
  },
  (table) => [
    index("idx_api_key_key").using("btree", table.key.asc().nullsLast()),
    index("idx_api_key_open_user_id").using(
      "btree",
      table.openUserId.asc().nullsLast()
    ),
  ]
);

File: packages/go-shared/pkg/model/api_key.go
package model

import (
	"time"
)

type ApiKey struct {
	ID         int32      `gorm:"column:id;type:integer;primaryKey;autoIncrement:true" json:"id"`
	Name       string     `gorm:"column:name;type:character varying(255);not null" json:"name"`
	OpenUserID string     `gorm:"column:open_user_id;type:character varying(32);not null;index:idx_api_key_open_user_id,priority:1" json:"open_user_id"`
	Key        string     `gorm:"column:key;type:character varying(255);not null;index:idx_api_key_key,priority:1" json:"key"`
	Status     int32      `gorm:"column:status;type:integer;not null" json:"status"`
	CreatedAt  *time.Time `gorm:"column:created_at;type:timestamp with time zone;default:now()" json:"created_at"`
	ExpiresAt  *time.Time `gorm:"column:expires_at;type:timestamp with time zone" json:"expires_at"`
}

func (ApiKey) TableName() string {
	return "api_key"
}
